### PR TITLE
Fix: redact signature/creds values on printing

### LINKED ISSUE #563: Do not expose credentials and signature in requests trace and exception messages
Replace access key and signature with '*EXPOSED*' in both exception description and http tracing.

### PR DESCRIPTION
**Pull request by vadmeste**
Access key and signature were exposed in Exception description
and http tracing. This commit hides them.

Fixes #563

### CODE REVIEW

File: api/src/main/java/io/minio/errors/ErrorResponseException.java [hunk 1]
```diff
@@ -57,7 +57,9 @@ public String toString() {
         + "request={"
         + "method=" + request.method() + ", "
         + "url=" + request.httpUrl() + ", "
-        + "headers=" + request.headers()
+        + "headers=" + request.headers().toString()
+                .replaceAll("Signature=([0-9a-f]+)", "Signature=*REDACTED*")
+                .replaceAll("Credential=([^/]+)", "Credential=*REDACTED*")
```
Comment (balamurugana) [inline api/src/main/java/io/minio/errors/ErrorResponseException.java]: minio-java doesn't add `Credential` header on request.
What does S3 call return `Credential` header?

Comment (harshavardhana) [inline api/src/main/java/io/minio/errors/ErrorResponseException.java]: Credential is part of the Authorization header @balamurugana . This while making the request not what s3 sent back.

Comment (balamurugana) [inline api/src/main/java/io/minio/errors/ErrorResponseException.java]: Got it.  We would need to change the commit log like
```
fix: redact signature/credential values in request headers on printing
```

Comment (harshavardhana) [inline api/src/main/java/io/minio/errors/ErrorResponseException.java]: :+1:

**Pull request merged**